src/core/device.c

```diff
@@ -90,6 +97,37 @@ static int device_set_sysfs(Device *d, const char *sysfs) {
         return 0;
 }
 
+static void io_cost_qos_reset(IOCostQos *io_cost_qos) {
+        assert(io_cost_qos);
+
+        *io_cost_qos = (IOCostQos) {
+                            .ctrl = _IO_COST_CTRL_INVALID,
+        };
+        /* If either min or max is set to a value below 1
+         * the kernel will set to 1 */
```

Comment:
[Don't](https://en.wikipedia.org/wiki/Robustness_principle) rely on that.
Also, shouldn't the default be something like 100?

Reply:
The default that the kernel sets is 1 for both values if nothing is input.

Reply:
"nothing is input" means `min=...` is omitted or `min=0`?
Yeah, if there's a kernel default then it's simpler to not define a new within systemd.

Reply:
Both `min=...' and `min=0'. 

It would be 1%.

Reply:
If the systemd user doesn't specify `IOCostQOSMin=` or `IOCostQOSMax=` (in the current implementation), `min=0 max=0` is passed to the kernel and the range becomes `(min=1, max=1)` that is IMO counter-intuitive (since it effectively grounds QOS scaling). `(min=1, max=100000)` or `(min=100, max=100)` is more sensible IMO.
Additionally, this breaks if kernel stops accepting `min=0 max=0` as valid input.

Reply:
I'm dense. Where do you get the info that kernel defaults to `(min=1, max=1)` from? Strictly speaking, the docs [doesn't specify](https://www.kernel.org/doc/html/v5.17/admin-guide/cgroup-v2.html#io-interface-files) defaults for these two and as I understand the code the qos scaling range is initialized to `(min=1, max=10000)`.